src/synthesizer/csharp/mod.rs

```diff
@@ -0,0 +1,668 @@
+            stack_props_class.newline();
+        }
+
+        namespace.newline();
+
+        // Description - comment before the stack class
+        if let Some(descr) = ir.description {
+            namespace.line("/// <summary>");
+            for description_line in descr.split('\n') {
+                namespace.line(format!("/// {description_line}"));
+            }
+            namespace.line("/// </summary>");
+        }
+
+        // Stack class definition
+        let stack_class = namespace.indent_with_options(IndentOptions {
+            indent: INDENT,
+            leading: Some(format!("public class {stack_name} : Stack\n{{").into()),
+            trailing: Some("}".into()),
+            trailing_newline: true,
+        });
+
+        // Properties for each output
+        for output in &ir.outputs {
+            if let Some(description) = &output.description {
+                stack_class.line("/// <summary>");
+                for description_line in description.split('\n') {
+                    stack_class.line(format!("/// {description_line}"));
+                }
+                stack_class.line("/// </summary>");
+            }
+            stack_class.line(format!("public object {} {{ get; }} ", output.name));
+            stack_class.newline();
+        }
+
+        // Constructor
+        let ctor = stack_class.indent_with_options(IndentOptions {
+            indent: INDENT,
+            leading: Some(format!(
+                "public {}(Construct scope, string id, {}Props props = null) : base(scope, id, props)\n{{",
+                stack_name,
+                stack_name
+            ).into()),
+            trailing: Some("}".into()),
+            trailing_newline: true,
+        });
+
+        // Default values for input props
+        let have_default_or_special_type_params = &ir
+            .constructor
+            .inputs
+            .iter()
+            .filter(|p| p.constructor_type.contains("AWS::") || p.default_value.is_some())
+            .collect::<Vec<&ConstructorParameter>>();
+        if !have_default_or_special_type_params.is_empty() {
+            ctor.line("// Applying default props");
+            for param in have_default_or_special_type_params {
+                let name = pascal_case(&param.name);
+                // example: AWS::EC2::Image::Id, List<AWS::EC2::VPC::Id>, AWS::SSM::Parameter::Value<List<String>>
+                if param.constructor_type.contains("AWS::") {
+                    let value_as = match &param.constructor_type {
+                        t if t.contains("List") => "ValueAsList",
+                        _ => "ValueAsString",
+                    };
+                    let cfn_param = ctor.indent_with_options(IndentOptions {
+                        indent: INDENT,
+                        leading: Some(
+                            format!(
+                                "props.{name} = new CfnParameter(this, \"{name}\", new CfnParameterProps\n{{")
```

Comment:
interesante. I thought in typescript we change parameters -> props (as that's the `cdk way`. Is that not what we are doing in all languages?

Reply:
This is for parameters that use one of the "special" types like AWS::SSM::etc... The value still has to be retrieved by cloudformation from ssm, so we have to generate the CfnParameter in those cases.